### DISCUSSION TITLE
Unable to compile 1.3.3 myself/Erroneous VSC behavior please help

### DISCUSSION THREAD

**Discussion by DeathDonky** [Jyers/Marlin#932] [Q&A] [answered]
Let me start by saying I can compile v1.3.2 all day long with no issues. The moment I pull up the 1.3.3 source code to edit for my needs VSC keeps areas greyed out that arent commented and as far as I know shouldn't be and im not sure why? for example the PID section under line 514(screenshot) isn't greyed out or darkened on the v1.3.2 that I edit and is on v1.3.3 and i have them identical in this area. 

Furthermore when I load up the V1.3.3 code into VSC (this is with and without replacing the config files with the preconfigured ones I use as "starting points") it gives me squiggle errors under all the "ENABLE" text. Mentions something about an "include path" and Im sorry but im just really lost here. Love the work you guys do! would appreciate the help. 
![PID Screen](https://user-images.githubusercontent.com/83191001/120077568-97c3e500-c070-11eb-9aa8-5291a4ae97e9.PNG)

**Comment by GordinhoPl** [comment]
I have the same issue, but it is only IntelliSense issue, so enabled function will remain grayed, but you can still compile Marlin and it will work. You only need to pay more attention, which oarts of code will be compiled ;)

**Reply by DeathDonky** [reply]
Should I also “ignore” the include path thing it’s giving me? I haven’t tried to “build” this version as the grayed out bits had me a little nervous but I’m kinda new to this whole coding thing.

**Comment by Jyers** [comment]
Had this issue myself a while ago. Never found a fix I was super happy with, but I did find a workaround #723

**Comment by Quint097** [comment, marked as answer]
I think I might have stumbled on a solution to this problem.

I did the following, which solved the problem for me:

- I got to the settings by going to **File** > **Preferences** > **Setting**
- I opened the extension settings and scroll down to **PlatformIO IDE configuration**
- I enabled the **Use Development PIOCore** setting, waited for it to safe and then restarted VS Code.
- When VS Code reopened it started reinstalling PlatformIO IDE. I waited for that to finish.
- I closed all open folders and then reopened the folder that didn't want to be recognised by PlatformIO.

It now recognised the project correctly and it started generating the correct files in the `.vscode` folder.
however it didn't solve the problem initially. I had to open one of the files showing errors, click on **PlatformIO** in the bottom right corner and then click on **Edit Configuration (JSON)** in the top bar that popped up. This opens `.vscode\c_cpp_properties.json` but i didn't need to do anything there because when i selected the previous file, the errors were gone.

since making these changes, I haven't had anymore problems or errors.

**Reply by DeathDonky** [reply]
This worked for 100% of the issues I was facing. No more squiggle errors and the areas that should and shouldn't be darkened seem to be corrected. THANK YOU SO F***** MUCH!

**Reply by Quint097** [reply]
You're welcome! I'm glad that I could help.

**Reply by Jyers** [reply]
Oh my lord! Not only does this mean that I don't have to do my weird workaround anytime I clone my repo, but on top of that this fixed another weird squiggle issue I was having! For like a month now I have been dealing with every file I edit having two errors in the include paths due to my TMC driver library. Do you know how annoying it is to ALWAYS have two errors in every file you visit??? Its so small but it really gets to you.

I don't know from what holy reservoir you pulled this information but I can not thank you enough!

**Reply by Quint097** [reply]
I sort of stumbled across it by accident.
I was having these problems for several weeks now and it became increasingly more difficult to write code without the help of IntelliSense. I'd looked online but couldn't really find anything and I'd also seen your discussion about it from a month ago, which didn't really get any further either. 
After trying just about everything I could think of, I started to change any settings related to PlatformIO in the hope that would do anything. which it suddenly did.

I later did a search for this specific setting and found one possible explanation for way this works, here: https://docs.platformio.org/en/latest/core/installation.html#development-version

**Reply by Cal-LF** [reply]
Thank you for sharing this treasure of info. Works with no errors now.
**Can we pin this somewhere so others can find this fix easily?**
